### ISSUE TITLE
Missing parent and child modular records (round 162976149)

### ISSUE THREAD

**Issue by mhess-swl** [hashgraph/hedera-services#12875]
### Description

A `ScheduleSign` transaction submitted by a user, _and_ an additional scheduled transaction executed after the schedule sign – NOT because of the parent schedule sign, as far as I know – results in the parent and child records completely missing from the modular record stream. The mod code reports the scheduled transaction erroring with `FAIL_INVALID`, specifically because of a non-zero-sum hbar exchange.

Parent Hashscan transaction [here](https://hashscan.io/mainnet/transaction/1708386587.941714630)
Scheduled Hashscan transaction [here](https://hashscan.io/mainnet/transactionsById/0.0.688593-1708386431-494212732)

**Parent transaction body:**
```
transactionID {
  transactionValidStart {
    seconds: 1708386579
    nanos: 311984902
  }
  accountID {
    accountNum: 726936
  }
}
nodeAccountID {
  accountNum: 27
}
transactionFee: 500000000
transactionValidDuration {
  seconds: 120
}
scheduleSign {
  scheduleID {
    scheduleNum: 4722866
  }
}
```

Modular exception stacktrace:
```
0 = {StackTraceElement@16103} "com.hedera.node.app.service.token.impl.RecordFinalizerBase.hbarChangesFrom(RecordFinalizerBase.java:88)"
1 = {StackTraceElement@16104} "com.hedera.node.app.service.token.impl.handlers.FinalizeParentRecordHandler.finalizeParentRecord(FinalizeParentRecordHandler.java:112)"
2 = {StackTraceElement@16105} "com.hedera.node.app.workflows.handle.HandleWorkflow.handleUserTransaction(HandleWorkflow.java:630)"
3 = {StackTraceElement@16106} "com.hedera.node.app.workflows.handle.HandleWorkflow.handlePlatformTransaction(HandleWorkflow.java:307)"
4 = {StackTraceElement@16107} "com.hedera.node.app.workflows.handle.HandleWorkflow.handleRound(HandleWorkflow.java:273)"
5 = {StackTraceElement@16108} "com.hedera.node.app.Hedera.onHandleConsensusRound(Hedera.java:957)"
6 = {StackTraceElement@16109} "com.hedera.node.app.state.HederaLifecyclesImpl.onHandleConsensusRound(HederaLifecyclesImpl.java:63)"
7 = {StackTraceElement@16110} "com.hedera.node.app.state.merkle.MerkleHederaState.handleConsensusRound(MerkleHederaState.java:328)"
8 = {StackTraceElement@16111} "com.swirlds.platform.recovery.EventRecoveryWorkflow.applyTransactions(EventRecoveryWorkflow.java:479)"
9 = {StackTraceElement@16112} "com.swirlds.platform.recovery.EventRecoveryWorkflow.handleNextRound(EventRecoveryWorkflow.java:391)"
10 = {StackTraceElement@16113} "com.swirlds.platform.recovery.EventRecoveryWorkflow.reapplyTransactions(EventRecoveryWorkflow.java:330)"
11 = {StackTraceElement@16114} "com.swirlds.platform.recovery.EventRecoveryWorkflow.recoverState(EventRecoveryWorkflow.java:166)"
12 = {StackTraceElement@16115} "com.swirlds.platform.cli.EventStreamRecoverCommand.call(EventStreamRecoverCommand.java:133)"
13 = {StackTraceElement@16116} "com.swirlds.platform.cli.EventStreamRecoverCommand.call(EventStreamRecoverCommand.java:37)"
14 = {StackTraceElement@16117} "picocli.CommandLine.executeUserObject(CommandLine.java:1953)"
15 = {StackTraceElement@16118} "picocli.CommandLine.access$1300(CommandLine.java:145)"
16 = {StackTraceElement@16119} "picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)"
17 = {StackTraceElement@16120} "picocli.CommandLine$RunLast.handle(CommandLine.java:2352)"
18 = {StackTraceElement@16121} "picocli.CommandLine$RunLast.handle(CommandLine.java:2314)"
19 = {StackTraceElement@16122} "picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)"
20 = {StackTraceElement@16123} "picocli.CommandLine$RunLast.execute(CommandLine.java:2316)"
21 = {StackTraceElement@16124} "picocli.CommandLine.execute(CommandLine.java:2078)"
22 = {StackTraceElement@16125} "com.swirlds.cli.PlatformCli.main(PlatformCli.java:195)"
```

### Steps to reproduce

Re-run transactions from round 162976149

### Hedera network

other

### Version

release/0.49, commit `3b125dd`

### Operating system

macOS

**Comment by mhess-swl** [comment]
# Comparison of Mono and Mod changes

## Mono 'Pending' Changes
### hbar and token
	0 = {BalanceChange@16342} "BalanceChange{token=ℏ, account=0.0.688593, alias=, units=64000, expectedDecimals=-1}"
	1 = {BalanceChange@16343} "BalanceChange{token=ℏ, account=0.0.726936, alias=, units=-100000, expectedDecimals=-1}"
	2 = {BalanceChange@16344} "BalanceChange{nft=0.0.590742, serialNo=6, from=0.0.688593, to=0.0.726936, counterPartyAlias=}"
	3 = {BalanceChange@16345} "BalanceChange{nft=0.0.639710, serialNo=39, from=0.0.688593, to=0.0.726936, counterPartyAlias=}"
	4 = {BalanceChange@16346} "BalanceChange{nft=0.0.908274, serialNo=60, from=0.0.688593, to=0.0.726936, counterPartyAlias=}"
	5 = {BalanceChange@16347} "BalanceChange{nft=0.0.706117, serialNo=108, from=0.0.688593, to=0.0.726936, counterPartyAlias=}"
	6 = {BalanceChange@16348} "BalanceChange{nft=0.0.833886, serialNo=186, from=0.0.688593, to=0.0.726936, counterPartyAlias=}"
	7 = {BalanceChange@16349} "BalanceChange{nft=0.0.833886, serialNo=218, from=0.0.688593, to=0.0.726936, counterPartyAlias=}"
	8 = {BalanceChange@16350} "BalanceChange{nft=0.0.1029854, serialNo=284, from=0.0.688593, to=0.0.726936, counterPartyAlias=}"
	9 = {BalanceChange@16351} "BalanceChange{token=ℏ, account=0.0.580888, alias=, units=5000, expectedDecimals=-1}"
	10 = {BalanceChange@16352} "BalanceChange{token=ℏ, account=0.0.148668, alias=, units=5000, expectedDecimals=-1}"
	11 = {BalanceChange@16353} "BalanceChange{token=ℏ, account=0.0.688280, alias=, units=12000, expectedDecimals=-1}"
	12 = {BalanceChange@16354} "BalanceChange{token=ℏ, account=0.0.610300, alias=, units=6000, expectedDecimals=-1}"
	13 = {BalanceChange@16355} "BalanceChange{token=ℏ, account=0.0.744081, alias=, units=1000, expectedDecimals=-1}"
	14 = {BalanceChange@16356} "BalanceChange{token=ℏ, account=0.0.732864, alias=, units=7000, expectedDecimals=-1}"
### implied transfers
	0 = {FcAssessedCustomFee@16846} "FcAssessedCustomFee{token=ℏ, account=EntityId{shard=0, realm=0, num=580888}, units=5000, effective payer accounts=[688593]}"
	1 = {FcAssessedCustomFee@16847} "FcAssessedCustomFee{token=ℏ, account=EntityId{shard=0, realm=0, num=148668}, units=5000, effective payer accounts=[688593]}"
	2 = {FcAssessedCustomFee@16848} "FcAssessedCustomFee{token=ℏ, account=EntityId{shard=0, realm=0, num=688280}, units=12000, effective payer accounts=[688593]}"
	3 = {FcAssessedCustomFee@16849} "FcAssessedCustomFee{token=ℏ, account=EntityId{shard=0, realm=0, num=610300}, units=6000, effective payer accounts=[688593]}"
	4 = {FcAssessedCustomFee@16850} "FcAssessedCustomFee{token=ℏ, account=EntityId{shard=0, realm=0, num=744081}, units=1000, effective payer accounts=[688593]}"
	5 = {FcAssessedCustomFee@16851} "FcAssessedCustomFee{token=ℏ, account=EntityId{shard=0, realm=0, num=732864}, units=7000, effective payer accounts=[688593]}"

## Mod Changes
### `hbarChanges` (size = 10)
```
{AccountID@15311} "AccountID[account=OneOf[kind=ACCOUNT_NUM, value=800]]" -> -365478 
{AccountID@15313} "AccountID[account=OneOf[kind=ACCOUNT_NUM, value=98]]" -> 73835 
{AccountID@15309} "AccountID[account=OneOf[kind=ACCOUNT_NUM, value=726936]]" -> -182038 
{AccountID@15305} "AccountID[account=OneOf[kind=ACCOUNT_NUM, value=688593]]" -> 64000
{AccountID@15307} "AccountID[account=OneOf[kind=ACCOUNT_NUM, value=580888]]" -> 308708 
{AccountID@15303} "AccountID[account=OneOf[kind=ACCOUNT_NUM, value=148668]]" -> 5000
null entry < missing 688280 account? > 
null entry < missing 610300 account? > 
{AccountID@15315} "AccountID[account=OneOf[kind=ACCOUNT_NUM, value=744081]]" -> 1000
{AccountID@15317} "AccountID[account=OneOf[kind=ACCOUNT_NUM, value=732864]]" -> 7000
``` `prepaidRewards`
"AccountID[shardNum=0, realmNum=0, account=OneOf[kind=ACCOUNT_NUM, value=580888]]" -> 303708
"AccountID[shardNum=0, realmNum=0, account=OneOf[kind=ACCOUNT_NUM, value=688280]]" -> 69973

### `tokenChanges` (size = 6)
```
{TokenID@15366} "TokenID[shardNum=0, realmNum=0, tokenNum=706117]" -> {ArrayList@15367}  size = 1
{TokenID@15368} "TokenID[shardNum=0, realmNum=0, tokenNum=590742]" -> {ArrayList@15369}  size = 1
{TokenID@15370} "TokenID[shardNum=0, realmNum=0, tokenNum=639710]" -> {ArrayList@15371}  size = 1
{TokenID@15372} "TokenID[shardNum=0, realmNum=0, tokenNum=1029854]" -> {ArrayList@15373}  size = 1
null entry
null entry
```

## Summary Of Hbar Differences
| Account | Mono (Expected) | Modular Diffs |
|--------|--------|--------|
| 800 | -365478 | not explicitly listed in mono changes |
| 98 |  73835 | not explicitly listed in mono changes |
| 726936 | -182038 | -100000
| 688593 | 64000 ||
| 580888 | 308708	| 5000 + 303708 = 308708
| 148668 | 5000 ||
| 688280 | 12000 | 69973 |
| 610300 | 6000 | missing |

**Comment by mhess-swl** [comment]
Possibly related to [#12694](https://github.com/hashgraph/hedera-services/issues/12694)

**Comment by mhess-swl** [comment]
Verified this is fixed in `release/0.49` (commit `c6dab4c`)
